### PR TITLE
Fix Vtexplains StripComments

### LINKED ISSUE #5981: sql like  select * from t where name like '*//*'  in StripComments will lead to a deadloop and a cpu usge raise
https://github.com/vitessio/vitess/blob/490bac139a3dadb7838c7a282cf3e4ee879ffffb/go/vt/sqlparser/comments.go#L169
sql like  `select * from t where name like '*//*' ` in StripComments will lead to a deadloop and a cpu usge raise

### PR DESCRIPTION
**Pull request by saifalharthi**
Closes #5981
Signed-off-by: Saif Alharthi <saif@saifalharthi.me>

### CODE REVIEW

**Comment by harshit-gangal** [comment]
Issue related #5981

**Comment by saifalharthi** [comment]
Thanks @harshit-gangal. Updated the issue description

File: go/vt/sqlparser/comments.go [hunk 1]
```diff
@@ -186,6 +192,13 @@ func StripComments(sql string) string {
 	return sql
 }
 
+func containsQuotations(sql string, start, end int) bool {
+	if start-1 < 0 || end+2 >= len(sql) {
+		return false
+	}
+	return (sql[start-1] == '\'' && sql[end+2] == '\'') || (sql[start-1] == '"' && sql[end+2] == '"')
```
Comment (harshit-gangal) [inline go/vt/sqlparser/comments.go]: as start-1 and end+2 are used at 3 places. Can you place them under vairable and give them a meaningful name which can help understand this return condition?

**Comment by harshit-gangal** [comment]
I tried below case and got this
test input: 'select * from customer where name like 'abc*/la/*xyz''
got SQL -> select * from customer where name like 'abcxyz'
want -> select * from customer where name like 'abc*/la/*xyz'
is my expectation correct?

**Comment by saifalharthi** [comment]
@harshit-gangal Yes, your expectation is correct. I changed the code so it can use the parser. And the previous code if it is not. I am not sure if we should change the tests to display an error the string cannot be parsed. But this seems like a good compromise.

**Comment by harshit-gangal** [comment]
Let's switch this function to use splitMarginComments and also fix the current issues with that function.

**Comment by saifalharthi** [comment]
I discussed with @sougou about this issue and decided to remove PARTITION  so vtexplain can match vitess's current behavior.

**Pull request merged**